common/ferrostar/src/simulation.rs

```diff
@@ -125,33 +158,108 @@ pub fn advance_location_simulation(
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::algorithms::snap_user_location_to_line;
+    use geo::HaversineDistance;
+    use rstest::rstest;
 
-    #[test]
-    fn advance_to_next_location() {
-        let mut state = location_simulation_from_coordinates(vec![
-            GeographicCoordinate { lng: 0.0, lat: 0.0 },
-            GeographicCoordinate { lng: 1.0, lat: 1.0 },
-            GeographicCoordinate { lng: 2.0, lat: 2.0 },
-            GeographicCoordinate { lng: 3.0, lat: 3.0 },
-        ])
+    #[rstest]
+    #[case(None)]
+    #[case(Some(10.0))]
```

Comment:
This lets us parameterize tests with case variables. You can read this like "for each case, pass these parameters into the respective case annotated arguments."